core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -387,33 +409,19 @@ class WorkspaceService(protected val userInfo: UserInfo,
     //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
     //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
     //ExecutionContext run the futures whenever
-    val abortWorkflowFuture: Future[Seq[WorkflowRecord]] =
-      requesterPaysSetupService.revokeAllUsersFromWorkspace(workspaceContext).flatMap { _ =>
-        dataSource.inTransaction { dataAccess =>
-          for {
-            // Gather any active workflows with external ids
-            workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
-
-            //If a workflow is not done, automatically change its status to Aborted
-            _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { recs =>
-              recs.collect {
-                case wf if !WorkflowStatuses.withName(wf.status).isDone =>
-                  dataAccess.workflowQuery.updateStatus(wf, WorkflowStatuses.Aborted) { status =>
-                    if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceName, wf.submissionId))(status))
-                    else None
-                  }
-              }
-            }
-          } yield {
-            workflowsToAbort
-          }
+    val gatherWorkflowsFuture: Future[Seq[WorkflowRecord]] = gatherWorkflowsToAbortAndSetStatusToAborted(workspaceName, workspaceContext)
```

Comment:
Was there a conscious reason why this call to `gatherWorkflowsToAbortAndSetStatusToAborted` was chosen to be run outside of the `for` comp?  I know this happened before the `for` comp before, but looking at this now, I probably would have opted to put this call inside now.  

Functionally, as written now, `gatherWorkflowsToAbortAndSetStatusToAborted` will be started and then processing will immediately proceed (without waiting for the future to complete) into the `for` comp and start `revokeAllUsersFromWorkspace`, so that those two futures are running in parallel.  By moving this line inside the `for` comp, you would be lining them up to run in sequence, which is how it functioned before anyway.  And if we really wanted to keep things ordered the same, the `revokeAllUsersFromWorkspace` should be above the call to `gatherWorkflowsToAbortAndSetStatusToAborted`.  

So I've talked myself into it.  I would say move `gatherWorkflowsToAbortAndSetStatusToAborted` inside the `for` comp and make it second call after `revokeAllUsersFromWorkspace`.   Unless, of course, you had a conscious reason why you have it as written.

Reply:
yeah, the original code has `val deletionFuture` defined outside of the for comp, so I was just keeping it there.

I see your point about keeping `revokeAllUsersFromWorkspace` and `gatherWorkflowsToAbortAndSetStatusToAborted` in the same order. Looking at the git history though I don't think the order particularly matters
https://github.com/broadinstitute/rawls/pull/1335
https://broadworkbench.atlassian.net/browse/CA-1095

I can't start the for comp with an assignment (`=`), so I'll put it under `revokeAllUsersFromWorkspace`.